src/graphics.c

```diff
@@ -888,20 +877,13 @@ GdipDrawBeziers (GpGraphics *graphics, GpPen *pen, GDIPCONST GpPointF *points, I
 GpStatus WINGDIPAPI
 GdipDrawBeziersI (GpGraphics *graphics, GpPen *pen, GDIPCONST GpPoint *points, INT count)
 {
-	if (count == 0)
-		return Ok;
-
-	if (!graphics || !pen || !points)
+	GpPointF *pointsF;
+	
+	if (!points || count < 0)
 		return InvalidParameter;
 
-	switch (graphics->backend) {
-	case GraphicsBackEndCairo:
-		return cairo_DrawBeziersI (graphics, pen, points, count);
-	case GraphicsBackEndMetafile:
-		return metafile_DrawBeziersI (graphics, pen, points, count);
-	default:
-		return GenericError;
-	}
+	pointsF = convert_points (points, count);
+	return GdipDrawBeziers (graphics, pen, pointsF, count);
 }
```

Comment:
Memory leak?

Reply:
Good point. this applies to all of these cases

Reply:
Yep. The rest of the patch seems ok to me.